pkg/artifacts/signable.go

```diff
@@ -175,6 +197,61 @@ func ExtractOCIImagesFromResults(tr *v1beta1.TaskRun, logger *zap.SugaredLogger)
 	return objs
 }
 
+// ExtractIntotoSignableTargetFromResults extracts signable targets that aim to generate Intoto provenance as materials within TaskRun results and store them as StructuredSignable.
+func ExtractIntotoSignableTargetFromResults(tr *v1beta1.TaskRun, logger *zap.SugaredLogger) []interface{} {
+	ss := map[string]*StructuredSignable{}
+	var objs []interface{}
+	suffises := []string{"INTOTO_TARGET_NAME", "INTOTO_TARGET_DIGEST"}
```

Comment:
`INTOTO` doesn't really have anything to do with the input type, right? It's what we want the output type to be? How does this interact if users specify a different output format that isn't intoto?

Reply:
I think if users are trying to specify different output format, maybe additional fields need to be added, and maybe this struct can be reused for that purpose. This function is only used for intoto provenance generation though

Reply:
This is just a URI + digest though, right? I'm a bit hesitant to tie the naming of this directly to intoto, since it's not really intoto specific - intoto is just a way to format artifact data.

This also makes it easier to support other formats in the future (i.e. if slsa and intoto diverge).